video/out/vulkan/context_display.c

```diff
@@ -0,0 +1,423 @@
+            if (res != VK_SUCCESS) {
+                mp_warn(log, "      Failed enumerating plane displays\n");
+                continue;
+            }
+            planes_to_displays[j] = displays;
+        }
+
+        // Enumerate Displays and Modes
+        VkDisplayPropertiesKHR *props =
+            talloc_array(tmp, VkDisplayPropertiesKHR, num_displays);
+        res = vkGetPhysicalDeviceDisplayPropertiesKHR(device, &num_displays, props);
+        if (res != VK_SUCCESS) {
+            mp_warn(log, "    Failed enumerating display properties\n");
+            continue;
+        }
+
+        for (int j = 0; j < num_displays; j++) {
+            mp_info(log, "    Display %d: '%s' (%dx%d)\n",
+                    j,
+                    props[j].displayName,
+                    props[j].physicalResolution.width,
+                    props[j].physicalResolution.height);
+
+            VkDisplayKHR display = props[j].display;
+
+            mp_info(log, "    Modes:\n");
+
+            int num_modes = 0;
+            vkGetDisplayModePropertiesKHR(device, display, &num_modes, NULL);
+            if (!num_modes) {
+                mp_info(log, "      No available modes for display.\n");
+                continue;
+            }
+
+            VkDisplayModePropertiesKHR *modes =
+                talloc_array(tmp, VkDisplayModePropertiesKHR, num_modes);
+            res = vkGetDisplayModePropertiesKHR(device, display, &num_modes, modes);
+            if (res != VK_SUCCESS) {
+                mp_warn(log, "      Failed enumerating display modes\n");
+                continue;
+            }
+
+            for (int k = 0; k < num_modes; k++) {
+                mp_info(log, "      Mode %02d: %dx%d (%02d.%03d Hz)\n",
+                        k,
+                        modes[k].parameters.visibleRegion.width,
+                        modes[k].parameters.visibleRegion.height,
+                        modes[k].parameters.refreshRate / 1000,
+                        modes[k].parameters.refreshRate % 1000);
+            }
+
+            mp_info(log, "    Planes:\n");
+            for (int k = 0; k < num_planes; k++) {
+                VkDisplayKHR *displays = planes_to_displays[k];
+                for (int d = 0; displays[d]; d++) {
+                    if (displays[d] == display) {
+                        mp_info(log, "      Plane: %d\n", k);
+                    }
+                }
+                displays = NULL;
+            }
+        }
+    }
+
+done:
+    talloc_free(tmp);
+    vkDestroyInstance(inst, NULL);
+    return ret;
+}
+
+static int display_validate_spec(struct mp_log *log, const struct m_option *opt,
+                                struct bstr name, struct bstr param)
+{
+    if (bstr_equals0(param, "help")) {
+        print_display_info(log);
+        return M_OPT_EXIT;
+    }
+
+    char *spec = bstrto0(NULL, param);
+    if (!parse_display_spec(log, spec, NULL, NULL, NULL)) {
```

Comment:
Should the enumeration code from `print_display_info` be merged into this function so that you can also verify that the parsed integers correspond to really existing modes?